core/src/main/java/org/frankframework/configuration/AdapterManager.java

```diff
@@ -224,19 +235,17 @@ private void doClose() {
 			stop(); //Call this just in case...
 		}
 
-		while (!getStopAdapterThreads().isEmpty()) {
-			log.debug("waiting for stop threads to end: {}", () -> StringUtil.safeCollectionToString(getStopAdapterThreads()));
+		while (!stopAdapterThreads.isEmpty()) {
+			log.debug("waiting for stop threads to end: {}", () -> StringUtil.safeCollectionToString(stopAdapterThreads));
+
 			try {
 				Thread.sleep(1000);
 			} catch (InterruptedException e) {
 				log.warn("Interrupted thread while waiting for stop threads to end", e);
 			}
 		}
 
-		while (!getAdapterList().isEmpty()) {
-			Adapter adapter = getAdapter(0);
-			unRegisterAdapter(adapter);
-		}
+		adapters.values().stream().forEach(this::removeAdapter);
```

Comment:
Deze code veranderd de data waarover de stream itereert, tijdens de iteratie.

Die geeft een grote kans dat er een `ConcurrentModificationException` kan komen.

Misschien niet nu in onze huidige unit-tests of op de huidige JDK versies maar het is volgens mij niet gegarandeerd dat dit altijd goed blijft gaan.